src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+package Duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    public static String parse(String userInput, TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
This function is very long. Consider breaking it up into separate functions and adding comments. It will help with readability.